### PR TITLE
Security advisories for 3.5.0 and 2.28.5

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**

### CODE REVIEW

File: security-advisories/mbedtls-security-advisory-2023-10-1.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Buffer overread in TLS stream cipher suites
+
+**Title** |  Buffer overread in TLS stream cipher suites
+---|---
+**CVE** |  CVE-2023-43615
+**Date** |  05 October 2023
+**Affects** |  All versions of Mbed TLS
+**Impact** |  A remote attacker may cause a crash or information disclosure.
+**Severity** |  Medium
```
Comment (yanesca) [inline security-advisories/mbedtls-security-advisory-2023-10-1.md:9]: Why are we issuing a security advisory for a medium severity issue?

File: security-advisories/mbedtls-security-advisory-2023-10-2.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+**Affects** |  Mbed TLS 3.2.0 and above
+**Impact** |  A remote attacker may cause arbitrary code execution.
+**Severity** |  HIGH
+**Credit** |  OSS-Fuzz
+
+## Vulnerability
+
+A TLS 1.3 client or server configured with support for signature-based authentication (i.e. any non-PSK key exchange) is vulnerable to a heap buffer overflow. An unauthenticated malicious peer can overflow the TLS handshake structure by sending an overly long ECDH or FFDH public key.
+
+A TLS 1.2 server configured with `MBEDTLS_USE_PSA_CRYPTO` and with support for a cipher suite using ECDH and a signature is vulnerable to a heap buffer overflow. An unauthenticated malicious peer can overflow the TLS handshake structure by sending an overly long ECDH public key. TLS 1.2 clients, and builds with `MBEDTLS_USE_PSA_CRYPTO` are not affected.
```
Comment (yanesca) [inline security-advisories/mbedtls-security-advisory-2023-10-2.md:16]: I am not sure I understand this part. How can be
"A TLS 1.2 server configured with `MBEDTLS_USE_PSA_CRYPTO` and with support for a cipher suite using ECDH and a signature" vulnerable if "builds with `MBEDTLS_USE_PSA_CRYPTO` are not affected"?

Comment (tom-cosgrove-arm) [reply security-advisories/mbedtls-security-advisory-2023-10-2.md:16]: Grr, same problem in #1091 which I approved, reading the second `with` as `without` :(

File: security-advisories/mbedtls-security-advisory-2023-10-1.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+Affected users will want to upgrade to Mbed TLS 3.5.0 or 2.28.5 depending on the branch they're currently using.
+
+## Work-around
+
+The vulnerability is not present in the default build of Mbed TLS. It is only present if the compile-time configuration enables the vulnerable cipher suites. If you use a custom configuration and you want to check that the vulnerable cipher suites are not included in your build:
+
+* In Mbed TLS 3.x or 2.28, make sure that `MBEDTLS_CIPHER_NULL_CIPHER` is not enabled.
+* In Mbed TLS 2.28, also make sure that `MBEDTLS_REMOVE_ARC4_CIPHERSUITES` is enabled, or that `MBEDTLS_ARC4_C` is not enabled.
+
+If the vulnerable cipher suites are enabled at compile time, they can be disabled at compile time by calling `mbedtls_ssl_conf_ciphersuites()` with a list that does not include null-cipher or RC4 cipher suites. Alternatively, call `mbedtls_ssl_conf_ciphersuites_for_version()` for all affected protocol versions (SSLv3, TLS 1.0, TLS 1.1, TLS 1.2).
```
Comment (yanesca) [inline security-advisories/mbedtls-security-advisory-2023-10-1.md:35]:
```suggestion
If the vulnerable cipher suites are enabled at compile time, they can be disabled at run time by calling `mbedtls_ssl_conf_ciphersuites()` with a list that does not include null-cipher or RC4 cipher suites. Alternatively, call `mbedtls_ssl_conf_ciphersuites_for_version()` for all affected protocol versions (SSLv3, TLS 1.0, TLS 1.1, TLS 1.2).
```

File: security-advisories/mbedtls-security-advisory-2023-10-1.md (hunk 1 above)
Comment (gilles-peskine-arm) [reply security-advisories/mbedtls-security-advisory-2023-10-1.md:9]: We discussed the severity in a security team meeting IIRC. An unauthenticated peer can crash a TLS server on a typical platform, so we decided to issue a CVE.

Comment (yanesca) [reply security-advisories/mbedtls-security-advisory-2023-10-1.md:9]: Yes, I remember discussing it on slack, I think we agreed that it was on the top end of medium. My memories about the CVE are more blurry.

In the past, issuing a CVE meant that we have to publish an advisory on the website as well. This is not the case anymore, our new process doesn't mandate this. We can, if we want, but we don't have to.

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Review by yanesca — APPROVED** [review]
LGTM

**Pull request merged**